pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -525,6 +531,11 @@ def remove_content(self, **kwargs):
                 "_content_type_id",
                 Matcher.in_(type_ids),  # Criteria.with_field_in is deprecated
             )
+        # extend the criteria with the filters when available.
+        # search_for_criteria will fill out the PulpSearch#filters field with
+        # the rest of the content excluding type_ids
+        if filters is not None:
+            criteria = Criteria.and_(criteria, filters)
```

Comment:
Does this work if type_ids was not provided?  If so, criteria is going to be `None` at this point, so this will be equivalent to `Criteria.and_(None, filters)`.  Not sure and_ is designed to accept `None`.

I think that case should be tested - actually I think that should be the preferred usage, because Criteria already supports matching by type (Criteria.with_unit_type).  So with this change, the caller can use just the criteria object to achieve everything and there's no need to pass type_ids separately.

I believe the only reason type_ids was there is because it was required for the first removal use-case, but full Criteria support was not required yet when remove_content was introduced.  So it was added to use type_ids, but via *kwargs to ensure it could be deprecated at a later point when full criteria support was added as you're doing now.

Reply:
I have removed the filters and updated it with criteria. Criteria will be the one taken into account and will ignore type_ids kwarg if criteria is present.